src/components/tasks/card/CardTitleWrapper.tsx

```diff
@@ -0,0 +1,32 @@
+import { FC, ReactNode } from 'react';
+import Link from 'next/link';
+
+interface CardTitleWrapperProps {
+    children?: ReactNode;
+    condition: boolean;
+    to?: string;
+    taskId?: string;
+}
+
+export const CardTitleWrapper: FC<CardTitleWrapperProps> = ({
+    children,
+    condition,
+    to,
+    taskId,
+}) => {
+    if (!!condition && to) {
+        return (
+            <Link
+                href={{
+                    pathname: '/tasks/[id]',
+                }}
+                as={`/tasks/${taskId}`}
+                style={{ textDecoration: 'none' }}
+            >
+                {children}
+            </Link>
+        );
+    }
+
+    return <>{children}</>;
```

Comment:
why do we need to wrap the children component in a react fragment ?

Reply:
In order to avoid adding an extra node to the DOM, I have wrapped it in a Fragment.